### ISSUE TITLE
Server binary included in the package

### ISSUE THREAD

**Issue by rchl** [sublimelsp/LSP-yaml#1]
What's the reason for including binary in the package instead of using npm dependency directly?
https://github.com/RandomByte/LSP-yaml/blob/88fe49bd0eb96d6866e44afe25d4ca68e5f65fb7/yaml-language-server-resources/package.json#L7

**Comment by RandomByte** [comment]
I had issues with the v0.7.2 package available on npm. It just didn't gave any responses from some request types. I tested with the latest master and there it worked. I did some digging but could not immediately identify the relevant change. This is the relevant diff: https://github.com/redhat-developer/yaml-language-server/compare/0.7.2...7a66d53

So I figured to quickly package that working version until they do their next release, which I would then reference.

**Comment by rchl** [comment]
OK, but there is a better way to do it. You can just ship it in unpacked form.

That will make the diffs easy to see and also you don't need to package `node_modules` as those will be installed automatically by `lsp_utils`.

You can check for example https://github.com/sublimelsp/LSP-eslint/tree/master/vscode-eslint where we also don't use npm dependency directly for the same reason. We even have a script to pull changes from the repo to make updates easier.

**Comment by RandomByte** [comment]
Well, what I did was executing `npm run compile` followed by [`npm pack`](https://docs.npmjs.com/cli/pack) in the [yaml-language-server](https://github.com/redhat-developer/yaml-language-server/) project. This should give my a tarball identical to what would have been published to the npm registry if the project would have done a release for that commit.

But you are right, the tarball contains the `node_modules` of the server. However, so does the last version published to npm: https://registry.npmjs.org/yaml-language-server/-/yaml-language-server-0.7.2.tgz

Here's a pretty recent issue on that topic: https://github.com/redhat-developer/yaml-language-server/issues/253

I'll check whether we can just remove the `node_modules` as suggested in the issue.

I would like to stick to the tarball as it is identical to what would come from the npm registry. For example, this way the projects [`.npmignore`](https://github.com/redhat-developer/yaml-language-server/blob/master/.npmignore) is utilized.

**Comment by jfcherng** [comment]
I try to move from `LSP + redhat-developer/yaml-language-server` to `LSP-yaml` so I don't have to manage node deps by myself. But `LSP-yaml` is not working on my side. It just crash immediately after it starts. It seems not related to settings since I tried both the default shipped by this package and the previous settings I used in `LSP + redhat-developer/yaml-language-server`. 

```
lsp-yaml: internal/modules/cjs/loader.js:983
lsp-yaml: throw err;
:: --> lsp-yaml initialize(1): {'initializationOptions': {}, 'workspaceFolders': [{'uri': 'file:///E:/WWW/nthu-ee-implementation', 'name': 'nthu-ee-implementation'}], 'processId': 31712, 'clientInfo': {'name': 'Sublime Text LSP'}, 'rootUri': 'file:///E:/WWW/nthu-ee-implementation', 'capabilities': {'textDocument': {'rangeFormatting': {}, 'completion': {'completionItemKind': {'valueSet': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25]}, 'completionItem': {'deprecatedSupport': True, 'snippetSupport': True}}, 'declaration': {'linkSupport': True}, 'hover': {'contentFormat': ['markdown', 'plaintext']}, 'definition': {'linkSupport': True}, 'documentHighlight': {}, 'synchronization': {'willSaveWaitUntil': True, 'didSave': True, 'willSave': True}, 'references': {}, 'codeAction': {'codeActionLiteralSupport': {'codeActionKind': {'valueSet': []}}}, 'publishDiagnostics': {'relatedInformation': True}, 'rename': {}, 'colorProvider': {}, 'signatureHelp': {'signatureInformation': {'parameterInformation': {'labelOffsetSupport': True}, 'documentationFormat': ['markdown', 'plaintext']}}, 'typeDefinition': {'linkSupport': True}, 'formatting': {}, 'documentSymbol': {'symbolKind': {'valueSet': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26]}}, 'implementation': {'linkSupport': True}}, 'experimental': {}, 'workspace': {'applyEdit': True, 'executeCommand': {}, 'didChangeConfiguration': {}, 'symbol': {'symbolKind': {'valueSet': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26]}}, 'workspaceFolders': True, 'configuration': True}}, 'rootPath': 'E:\\WWW\\nthu-ee-implementation'}
```

**Comment by RandomByte** [comment]
Yes, thanks for the confirmation. I'm already looking into that. It seems that npm tries to resolve the package-lock.json entry for `"file:yaml-language-server-0.7.2.tgz",` against the CLIs cwd.

The reason I did not spot this earlier was that once npm successfully installed that tar, it is added to the cache and always found, now matter the cwd.

The only workaround I currently see is removing the `package-lock.json`

**Comment by RandomByte** [comment]
> The only workaround I currently see is removing the package-lock.json

PR: https://github.com/sublimelsp/LSP-yaml/pull/5

**Comment by RandomByte** [comment]
Released (both) fixes as [`v0.1.1`](https://github.com/sublimelsp/LSP-yaml/releases/tag/v0.1.1). Will try and test this on a *third* machine tomorrow :sweat_smile:

**Comment by RandomByte** [comment]
> Will try and test this on a *third* machine tomorrow 

Seems to be working :+1:
